x-pack/plugins/security_solution/public/common/components/hover_actions/index.tsx

```diff
@@ -216,6 +219,19 @@ export const HoverActions: React.FC<Props> = React.memo(
     );
 
     const isCaseView = scopeId === TimelineId.casePage;
+    const isTimelineView = scopeId === TimelineId.active;
+
+    const hideFilters = useMemo(
+      () => isAlertDetailsPage(pathname) && !isTimelineView,
+      [isTimelineView, pathname]
+    );
+
+    const hiddenActionsCount = useMemo(() => {
+      let count = 0;
```

Comment:
Consider something like the following to avoid re-assignment:

```ts
    const hiddenActionsCount = useMemo(() => {
      const hiddenTopNActions = hideTopN ? 1 : 0; // hides the `Top N` button
      const hiddenFilterActions = hideFilters ? 2 : 0; // hides both the `Filter In` and `Filter out` buttons

      return hiddenTopNActions + hiddenFilterActions;
    }, [hideFilters, hideTopN]);
```

Reply:
Will make this change although this function may not be necessary. Either way I'll leave it in as I think this paradigm is more extensible moving forward